### PR TITLE
Added validation which verifying integration command in layouts, playbooks or incidents fields

### PR DESCRIPTION
**Pull request by bziser**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/41246

## Description
Added validation which verifying that every integration command in layouts, playbooks or incidents fields relates to an existing command

## Screenshots
Paste here any images that will help the reviewer

## Must have
- [ ] Tests
- [ ] Documentation
- [ ] [VS Code Extension](https://docs.gitlab.com/ee/ci/yaml/#tags)
  - [ ] Functionality implemented in the extension - <u>\<link-to-pr-here\></u>
  - [ ] N/A - Functionality cannot be implemented in the extension because <u>\<add-reason-here\></u>

### CODE REVIEW

File: demisto_sdk/commands/common/hook_validations/id.py [hunk 1]
```diff
@@ -215,12 +215,9 @@ def _get_scripts_that_are_not_in_id_set(self, scripts_in_entity):
         # Ignore Builtin scripts because they are implemented on the server side and thus not in the id_set.json
         scripts_in_entity = self._remove_builtin_scripts(scripts_in_entity)
 
-        # Ignore integration commands scripts because they are not in the id_set.json
-        # Ignoring integration commands is temporary. Validate command should verify that each integration command
+        # Validate command should verify that each integration command
         # called from a layout, a layoutscontainer or an incident field is really exist.
```
Comment (yuvalbenshalom) [inline demisto_sdk/commands/common/hook_validations/id.py:219]:
```suggestion
        # called from a layout, a layoutscontainer or an incident field really exist in the content repo.
```

File: demisto_sdk/commands/common/hook_validations/id.py [hunk 2]
```diff
@@ -243,27 +240,29 @@ def _remove_builtin_scripts(self, scripts_set):
...
         """
-        For each script ID in the given scripts set checks if it is an integration command by checking if it contains
-        '|||'.  If a script is not an integration command add it to a new scripts set.
-        TODO: Ignoring integration commands is temporary. Validate command should verify that each integration command
-         called from a layout, a layoutscontainer or an incident field is really exist.
-         will be fixed in: https://github.com/demisto/etc/issues/41246
-
+        For each script ID in the given scripts set checks if it is an integration command by
+        checking if it contains '|||'.  If a script is  an integration command checks in id_set.json
+        if it exists.
```
Comment (yuvalbenshalom) [inline demisto_sdk/commands/common/hook_validations/id.py:247]: not sure I understand the name and the explanation. is this going thru scripts or playbooks?

File: demisto_sdk/commands/common/hook_validations/playbook.py [hunk 3]
```diff
@@ -314,20 +314,22 @@ def is_script_id_valid(self, id_set_file):
...
-                is_valid = self.check_script_name(task_script_name, id_set_scripts)
+            script_entry_to_check = script_id_used_in_task if script_id_used_in_task else task_script_name
+            integration_script_flag = "|||"
+            if script_id_used_in_task:
+                if integration_script_flag not in script_id_used_in_task:  # Checking script
+                    is_valid &= self.check_script_id(script_id_used_in_task, id_set_scripts)
+                else:  # Checking integration command
+                    is_valid &= self.check_integration_command(script_id_used_in_task,
+                                                               id_set_integrations)
+            if task_script_name and integration_script_flag not in task_script_name:
```
Comment (yuvalbenshalom) [inline demisto_sdk/commands/common/hook_validations/playbook.py:331]: need to add explanation to this case

File: demisto_sdk/commands/common/hook_validations/playbook.py [hunk 4]
```diff
@@ -346,6 +348,29 @@ def check_script_id(self, script_id_used_in_task, id_set_scripts):
         """
         return any([script_id_used_in_task in id_set_dict for id_set_dict in id_set_scripts])
 
+    def check_integration_command(self, integration_id_used_in_task, id_set_integrations,
+                                  empty_integration_id=True):
```
Comment (yuvalbenshalom) [inline demisto_sdk/commands/common/hook_validations/playbook.py:352]: maybe change empty_integration_id to command_without_brand? or something like that?

File: demisto_sdk/commands/common/hook_validations/playbook.py [hunk 5]
```diff
@@ -346,6 +348,29 @@ def check_script_id(self, script_id_used_in_task, id_set_scripts):
...
+        """
+        Checks if integration id and command exists in at least one of id_set's dicts
+        Args:
+            integration_id_used_in_task (str):  integration id from playbook
+            id_set_integrations (list): all integrations of id_set
+            empty_integration_id (bool): Whether integration_id is legal as None.
+             i.e.: |||Command is legal or not. true - legal, false - not
```
Comment (yuvalbenshalom) [inline demisto_sdk/commands/common/hook_validations/playbook.py:359]: the example us correct, but better explain that it excepts the case that the command does not include the brand/integration name.

**Review by yuvalbenshalom — CHANGES_REQUESTED** [review]
looks good,
see my comments. check if you an include an example for incident field with a triggered script.

File: demisto_sdk/commands/common/hook_validations/id.py (hunk 2 above)
Comment (bziser) [reply demisto_sdk/commands/common/hook_validations/id.py:247]: I changed it, hope it is more clear
It checks whether in the scripts_set are integration commands and if there are if they are legal

File: demisto_sdk/commands/common/hook_validations/playbook.py (hunk 4 above)
Comment (bziser) [reply demisto_sdk/commands/common/hook_validations/playbook.py:352]: You are right, it is better

File: demisto_sdk/commands/common/hook_validations/playbook.py (hunk 3 above)
Comment (bziser) [reply demisto_sdk/commands/common/hook_validations/playbook.py:331]: Added, check if it is better

**Review by yuvalbenshalom — APPROVED** [review]

**Pull request merged**